frontend/packages/kubevirt-plugin/src/components/modals/boot-order-modal/boot-order-modal.tsx

```diff
@@ -0,0 +1,93 @@
+import * as React from 'react';
+import * as _ from 'lodash';
+import { Modal, Button } from '@patternfly/react-core';
+import { ModalComponentProps } from '@console/internal/components/factory';
+import { k8sPatch } from '@console/internal/module/k8s';
+import { VMLikeEntityKind, BootableDeviceType } from '../../../types';
+import { getVMLikeModel, getDevices } from '../../../selectors/vm';
+import { PatchBuilder, PatchOperation } from '../../../k8s/utils/patch';
+import { getVMLikePatches } from '../../../k8s/patches/vm-template';
+import { BootOrder } from '../../boot-order';
+import { DeviceType } from '../../../constants';
+
+const modalTitle = 'Virtual machine boot order';
+
+export const BootOrderModal = ({
+  vmLikeEntity,
+  isOpen,
+  setOpen,
+  title = modalTitle,
+}: BootOrderModalProps) => {
+  const [devices, setDevices] = React.useState<BootableDeviceType[]>(getDevices(vmLikeEntity));
+
+  // Update device list on vmLikeEntity change.
+  React.useEffect(() => {
+    const updatedDevices = getDevices(vmLikeEntity);
+
+    setDevices(updatedDevices);
+  }, [vmLikeEntity]);
+
+  const submit = async (event) => {
+    event.preventDefault();
+
+    // Filter disks and interfaces from devices list.
+    const disks = [
+      ..._.filter(devices, (source) => source.type === DeviceType.DISK).map(
+        (source) => source.value,
+      ),
+    ];
+    const interfaces = [
+      ..._.filter(devices, (source) => source.type === DeviceType.NIC).map(
+        (source) => source.value,
+      ),
+    ];
+
+    // Patch k8s.
+    const patches = [
+      new PatchBuilder('/spec/template/spec/domain/devices/disks')
+        .setOperation(PatchOperation.REPLACE)
+        .setValue(disks)
+        .build(),
+      new PatchBuilder('/spec/template/spec/domain/devices/interfaces')
+        .setOperation(PatchOperation.REPLACE)
+        .setValue(interfaces)
+        .build(),
+    ];
+    const promise = k8sPatch(
+      getVMLikeModel(vmLikeEntity),
+      vmLikeEntity,
+      getVMLikePatches(vmLikeEntity, () => patches),
+    );
+
+    promise.then(() => setOpen(false)); // eslint-disable-line promise/catch-or-return
+  };
+
+  const actions = [
+    <Button key="confirm" variant="primary" onClick={submit}>
+      Save
+    </Button>,
+    <Button key="cancel" variant="link" onClick={() => setOpen(false)}>
+      Cancel
+    </Button>,
+  ];
+
+  return (
+    <Modal
```

Comment:
we are missing promise progress and error here Shouldn't we align this with console dialogs?

Reply:
> Shouldn't we align this with console dialogs?

console dialogs also ignore promises ( or rather don't do anything with them ) and leave the catching to the user.

for example in [disk-modal-enhanced.tsx](https://github.com/openshift/console/blob/master/frontend/packages/kubevirt-plugin/src/components/modals/disk-modal/disk-modal-enhanced.tsx#L41) we use console dialog with the same logic as here:
wait for promise to fulfil => then close dialog => ignore errors. 

> missing promise progress

we wait for promise to fulfil before closing the dialog ... since we do not handle negative flows it's the best we can do, until we start catching errors/timeouts.

> and error here

yes, we should handle errors, this will also fix the promise progress thing :-)

Reply:
> > Shouldn't we align this with console dialogs?
> 
> console dialogs also ignore promises ( or rather don't do anything with them ) and leave the catching to the user.

I think this feature is in a lot of dialogs and is quite useful

> 
> for example in [disk-modal-enhanced.tsx](https://github.com/openshift/console/blob/master/frontend/packages/kubevirt-plugin/src/components/modals/disk-modal/disk-modal-enhanced.tsx#L41) we use console dialog with the same logic as here:
> wait for promise to fulfil => then close dialog => ignore errors.
> 


that is not true, disk-modal-enhanced covers all cases. Please see disk-modal and `withHandlePromise` wrapper with `handlePromise`. Can we use something like that?


> > missing promise progress
> 
> we wait for promise to fulfil before closing the dialog ... since we do not handle negative flows it's the best we can do, until we start catching errors/timeouts.
> 
> > and error here
> 
> yes, we should handle errors, this will also fix the promise progress thing :-)

+1 we should handle all the cases

Reply:
update:
a - wait for patch to finish or fail, on ok -> close modal.
b - on error, inform user and leave modal open.
c - on outside change, inform users, let user manually reload devices.

Reply:
can we rather use `withHandlePromise` like in all of our modals and a lot of console ones? 

- we get a design consistency
- error and progress handling for free